sklearn/metrics/_plot/cap_curve.py

```diff
@@ -0,0 +1,313 @@
+        ----------
+        y_true : array-like of shape (n_samples,)
+            True labels.
+
+        y_pred : array-like of shape (n_samples,)
+            Target scores, can either be probability estimates of the positive
+            class, confidence values, or non-thresholded measure of decisions
+            (as returned by “decision_function” on some classifiers).
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Sample weights.
+
+        pos_label : int, float, bool or str, default=None
+            The label of the positive class. When `pos_label=None`, if `y_true`
+            is in {-1, 1} or {0, 1}, `pos_label` is set to 1, otherwise an
+            error will be raised.
+
+        name : str, default=None
+            Name of CAP curve for labeling. If `None`, name will be set to
+            `"Classifier"`.
+
+        ax : matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is created.
+
+        **kwargs : dict
+            Additional keywords arguments passed to matplotlib `plot` function.
+
+        Returns
+        -------
+        display : :class:`~sklearn.metrics.CumulativeAccuracyDisplay`
+            Object that stores computed values.
+        """
+        # validate and prepare data
+        if pos_label is None:
+            pos_label = 1
+        if sample_weight is None:
+            sample_weight = np.ones_like(y_true, dtype=float)
+
+        # ensure y_true is boolean for positive class identification
+        y_bool = y_true == pos_label
+
+        # sort predictions and true values based on the predictions
+        sorted_indices = np.argsort(y_pred)[::-1]
+        y_true_sorted = y_bool[sorted_indices]
+        sample_weight_sorted = sample_weight[sorted_indices]
+
+        # compute cumulative sums for true positives and all cases
+        cumulative_true_positives = np.cumsum(y_true_sorted * sample_weight_sorted)
+        cumulative_total = np.cumsum(sample_weight_sorted)
+
+        viz = cls(
+            cumulative_true_positives=cumulative_true_positives,
+            cumulative_total=cumulative_total,
+            estimator_name=name,
+            pos_label=pos_label,
+        )
+
+        return viz.plot(
+            ax=ax,
+            name=name,
+            normalize_scale=normalize_scale,
+            plot_chance_level=plot_chance_level,
+            **kwargs,
+        )
+
+    @classmethod
+    def from_estimator(
+        cls,
+        estimator,
+        X,
+        y,
+        *,
+        sample_weight=None,
+        response_method="auto",
+        pos_label=None,
+        normalize_scale=False,
+        plot_chance_level=False,
+        name=None,
+        ax=None,
+        **kwargs,
+    ):
+        """Create the Cumulative Accuracy Profile.
+
+        This is also known as a Gain or Lift Curve for classification, and a Lorenz
+        curve for regression with a positively valued target.
```

Comment:
```suggestion
        This is also known as a Gain or Lift Curve, or the mirrored Lorenz
        curve of Machine Learning for regression with a positively valued target.
```

Reply:
Since we agree to leave the Lorenz curve / regression case for a follow-up PR I would defer this change for later. Speaking about regression on an class that is designed and document to only accept classifier predictions would be a source confusion.